pkg/client/leaderelectionconfig/config.go

```diff
@@ -42,7 +42,8 @@ func BindFlags(l *componentbaseconfig.LeaderElectionConfiguration, fs *pflag.Fla
 		"of a leadership. This is only applicable if leader election is enabled.")
 	fs.StringVar(&l.ResourceLock, "leader-elect-resource-lock", l.ResourceLock, ""+
 		"The type of resource object that is used for locking during "+
-		"leader election. Supported options are `endpoints` (default) and `configmaps`.")
+		"leader election. Supported options are 'endpoints', 'configmaps', "+
+		"'leases', 'endpointsleases' (default) and 'configmapsleases'.")
```

Comment:
remove the default text, flag help already outputs the default value, and the default can differ by component
```suggestion
		"'leases', 'endpointsleases' and 'configmapsleases'.")
```